sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/JacksonHttpSessionOAuth2AuthorizedClientRepository.java

```diff
@@ -60,16 +60,15 @@ public void removeAuthorizedClient(String clientRegistrationId, Authentication p
         Assert.hasText(clientRegistrationId, "clientRegistrationId cannot be empty");
         Assert.notNull(request, MSG_REQUEST_CANNOT_BE_NULL);
         Map<String, OAuth2AuthorizedClient> authorizedClients = this.getAuthorizedClients(request);
-        if (!authorizedClients.isEmpty()) {
-            if (authorizedClients.remove(clientRegistrationId) != null) {
-                if (!authorizedClients.isEmpty()) {
-                    request.getSession().setAttribute(AUTHORIZED_CLIENTS_ATTR_NAME,
-                        serializeOAuth2AuthorizedClientMap(authorizedClients));
-                } else {
-                    request.getSession().removeAttribute(AUTHORIZED_CLIENTS_ATTR_NAME);
-                }
+        if (!authorizedClients.isEmpty() && authorizedClients.remove(clientRegistrationId) != null) {
+            if (!authorizedClients.isEmpty()) {
```

Comment:
do you still need this check? I think this is always true?

Reply:
then 
```
request.getSession().removeAttribute(AUTHORIZED_CLIENTS_ATTR_NAME);
```
will not have change to execute?

Also curious about this. Do we miss something?

Reply:
@stliu 

> do you still need this check? I think this is always true?

Line 64 is necessary because line 63 have this code: `authorizedClients.remove(clientRegistrationId)`.

And this code is copied from here:

https://github.com/spring-projects/spring-security/blob/5.6.3/oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/HttpSessionOAuth2AuthorizedClientRepository.java#L67-L82

Reply:
We still need this check. It is NOT always true. @stliu